nio-impl/src/main/java/org/xnio/nio/QueuedNioTcpServer.java

```diff
@@ -401,14 +418,16 @@ public SocketAddress getLocalAddress() {
 
     public void suspendAccepts() {
         handle.suspend(0);
+        suspendedDueToWatermark = false;
```

Comment:
If it's already suspended due to watermark then I think suspend should be a no-op (other than setting a user-suspended flag) - the actual suspend status of the channel should be (suspended by user | suspended due to watermark).  IOW you can't override suspention due to connection count.